make/test.mk

```diff
@@ -259,56 +256,32 @@ else
-        sleep 1; \
-    done
+	@echo Using image ${IMAGE_NAME} and namespace ${NAMESPACE} for the repository ${REPO_NAME}
+ifeq ($(REPO_NAME),registration-service)
+	# registration service is not integrated with OLM yet, so deploy it directly
+	sed -e 's|REPLACE_IMAGE|${IMAGE_NAME}|g' ${E2E_REPO_PATH}/deploy/deployment_dev.yaml | oc apply -f -
 else
-	sed -e 's|REPLACE_IMAGE|${IMAGE_NAME}|g' ${E2E_REPO_PATH}/deploy/operator.yaml | oc apply -f -
-endif
-
-.PHONY: build-and-deploy-service
-build-and-deploy-service:
-# build service
-# when e2e tests are triggered from different repo - eg. as part of PR in host-operator repo - and the image of the deployment is (not) provided
-ifeq ($(SET_IMAGE_NAME),)
-	# check if it is running locally
-    ifeq ($(OPENSHIFT_BUILD_NAMESPACE),)
-        ifneq ($(IS_OS_3),)
-			# using OS 3 (assuming it's minishift) then build image
-			$(eval IMAGE_NAME := quay.io/${GO_PACKAGE_ORG_NAME}/${REPO_NAME}:${DATE_SUFFIX})
-			$(MAKE) -C ${REPO_PATH} build
-			docker build -f ${REPO_PATH}/build/Dockerfile -t ${IMAGE_NAME} ${REPO_PATH}
-        else
-			# using OS 4 then build and push image
-			$(eval IMAGE_NAME := quay.io/${QUAY_NAMESPACE}/${REPO_NAME}:${DATE_SUFFIX})
-			$(MAKE) -C ${REPO_PATH} build
-			docker build -f ${REPO_PATH}/build/Dockerfile -t ${IMAGE_NAME} ${REPO_PATH}
-			docker push ${IMAGE_NAME}
-        endif
+    ifeq ($(IS_OS_3),)
+		# it is not using OS 3 so we will install operator via CSV
+		$(eval CATALOGSOURCE_NAME := $(shell sed -e 's|REPLACE_IMAGE|${IMAGE_NAME}|g' ${E2E_REPO_PATH}/hack/deploy_csv.yaml | oc apply -f - | grep catalogsource | awk '{print $$1;}'))
+		$(eval SUBSCRIPTION_NAME := $(shell sed -e 's|REPLACE_NAMESPACE|${NAMESPACE}|g' ${E2E_REPO_PATH}/hack/install_operator.yaml | oc apply -f - | grep subscription | awk '{print $$1;}'))
+		while [[ -z `oc get sa ${REPO_NAME} -n ${NAMESPACE} 2>/dev/null` ]] || [[ -z `oc get crd kubefedclusters.core.kubefed.k8s.io 2>/dev/null` ]]; do \
+			if [[ $${NEXT_WAIT_TIME} -eq 300 ]]; then \
+			   echo "reached timeout of waiting for ServiceAccount ${REPO_NAME} to be available in namespace ${NAMESPACE} and CRD kubefedclusters.core.kubefed.k8s.io to be available in the cluster - see following info for debugging:"; \
+			   echo "================================ CatalogSource =================================="; \
+			   oc get ${CATALOGSOURCE_NAME} -n openshift-marketplace -o yaml; \
+			   echo "================================ CatalogSource Pod Logs =================================="; \
+			   CATALOGSOURCE_NAME=${CATALOGSOURCE_NAME}; \
+			   oc logs `oc get pods -n openshift-marketplace -o name | grep $${CATALOGSOURCE_NAME#*/}` -n openshift-marketplace; \
+			   echo "================================ Subscription =================================="; \
+			   oc get ${SUBSCRIPTION_NAME} -n ${NAMESPACE} -o yaml; \
+			   $(MAKE) print-operator-logs REPO_NAME=${REPO_NAME} NAMESPACE=${NAMESPACE}; \
```

Comment:
do we need this commented out code?

Reply:
it's not commented out - it's just wrong formatting in GH